espflash/src/targets/flash_target/esp32.rs

```diff
@@ -48,6 +48,63 @@ impl FlashTarget for Esp32Target {
             connection.command(command)
         })?;
 
+        if !self.use_stub && connection.get_usb_pid()? == USB_SERIAL_JTAG_PID {
```

Comment:
Would you mind please adding a comment as to _why_ this is necessary, so that it doesn't get accidentally removed again in the future?

Reply:
Done :)